proof/refine/X64/Syscall_R.thy

```diff
@@ -400,7 +400,7 @@ lemma performInvocation_corres:
   "\<lbrakk> inv_relation i i'; call \<longrightarrow> block \<rbrakk> \<Longrightarrow>
    corres (dc \<oplus> (=))
      (einvs and valid_invocation i
-            and simple_sched_action
+            and schact_is_rct
```

Comment:
not very important, but I'm curious: any thoughts on why this (and related) change? I'm trying to remember whether the `simple_sched_action` was actually thought of as the minimum needed, or copy-pasted from elsewhere

Reply:
All the changes for the scheduler action in later files come down to the beefed up guards in `deleteObjects_corres`. I was perhaps slightly lazy and simply went along with what's on the `rt` branch here, namely, `schact_is_rct`. It seems like I could actually weaken it to `simple_sched_action`. With nothing in Refine or CRefine arch split it's a nightmare to go and change things like this, but I can go back and weaken things

Reply:
I'll buy the argument that it reduces the diff to `rt`. There are many preconditions we have that aren't strictly weakest preconditions, so it's not a significant effect. Thanks for explaining.